spotinst_sdk2/models/ocean/aws/__init__.py

```diff
@@ -1,370 +1,918 @@
+        self.auto_headroom_percentage = auto_headroom_percentage
+        self.enable_automatic_and_manual_headroom = enable_automatic_and_manual_headroom
+        self.extended_resource_definitions = extended_resource_definitions
 # endregion
 
+
 # region Capacity
 class Capacity:
-	"""
-	# Arguments
-	minimum: int
-	maximum: int
-	target: int
-	"""
-	def __init__(
-		self,
-		minimum=none,
-		maximum=none,
-		target=none):
-
-		self.minimum = minimum
-		self.maximum = maximum
-		self.target = target
+    """
+    # Arguments
+    minimum: int
+    maximum: int
+    target: int
+    """
+    def __init__(
+            self,
+            minimum: int = none,
+            maximum: int = none,
+            target: int = none):
+        self.minimum = minimum
+        self.maximum = maximum
+        self.target = target
 # endregion
 
+
 # region Strategy
+class AvailabilityVsCost(Enum):
+    cost_oriented = "costOriented"
+    balanced = "balanced"
+    cheapest = "cheapest"
+
+
+class ClusterOrientation:
+    """
+    # Arguments
+    availability_vs_cost: AvailabilityVsCost
+    """
+    def __init__(
+            self,
+            availability_vs_cost: AvailabilityVsCost = none):
+        self.availability_vs_cost = availability_vs_cost
+
+
+class SpreadNodesBy(Enum):
+    vcpu = "vcpu"
+    count = "count"
+
+
 class Strategy:
-	"""
-	# Arguments
-	utilize_reserved_instances: bool
-	fallback_to_od: bool
-	spot_percentage: int
-	"""
-	def __init__(
-		self,
-		utilize_reserved_instances=none,
-		fallback_to_od=none,
-		spot_percentage=none):
-
-		self.utilize_reserved_instances = utilize_reserved_instances
-		self.fallback_to_od = fallback_to_od
-		self.spot_percentage = spot_percentage
+    """
+    # Arguments
+    utilize_reserved_instances: bool
+    fallback_to_od: bool
+    spot_percentage: int
+    grace_period: int
+    draining_timeout: int
+    utilize_commitments: bool
+    cluster_orientation: ClusterOrientation
+    spread_nodes_by: SpreadNodesBy
+    """
+    def __init__(
+            self,
+            utilize_reserved_instances: bool = none,
+            fallback_to_od: bool = none,
+            spot_percentage: int = none,
+            grace_period: int = none,
+            draining_timeout: int = none,
+            utilize_commitments: bool = none,
+            cluster_orientation: ClusterOrientation = none,
+            spread_nodes_by: SpreadNodesBy = none):
+        self.utilize_reserved_instances = utilize_reserved_instances
+        self.fallback_to_od = fallback_to_od
+        self.spot_percentage = spot_percentage
+        self.grace_period = grace_period
+        self.draining_timeout = draining_timeout
+        self.utilize_commitments = utilize_commitments
+        self.cluster_orientation = cluster_orientation
+        self.spread_nodes_by = spread_nodes_by
 # endregion
 
+
+# region Scheduling
+class ClusterRoll:
+    """
+    # Arguments
+    batch-min_healthy_percentage: int
```

Comment:
batch-min_healthy_percentage to be changed to batch_min_healthy_percentage

Reply:
Done